mat64/io.go

```diff
@@ -0,0 +1,60 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mat64
+
+import (
+	"encoding/binary"
+	"io"
+)
+
+var (
+	littleEndian  = binary.LittleEndian
+	bigEndian     = binary.BigEndian
+	defaultEndian = littleEndian
+)
+
+func writeBin(w io.Writer, data interface{}) error {
+	return binary.Write(w, defaultEndian, data)
+}
+
+func writeInt(w io.Writer, n int) error {
+	return writeBin(w, int64(n))
+}
+
+func writeSlice(w io.Writer, data []float64) error {
+	var err error
+
+	for _, v := range data {
+		err = writeBin(w, v)
+		if err != nil {
+			return err
+		}
+	}
+
+	return err
+}
+
+func readBin(r io.Reader, data interface{}) error {
```

Comment:
See comment on writeBin. One line function that doesn't help much. Just put in place in the main function.